cdci_data_analysis/analysis/products.py

```diff
@@ -84,12 +85,13 @@ def set_analysis_parameters(self, query_dict):
     def set_api_code(self, query_dict, url=None):
         self.prod_dictionary['api_code'] = DispatcherAPI.set_api_code(query_dict, url=url)
 
-    def dump_analysis_parameters(self,work_dir,query_dict):
-        file_path=FilePath(file_dir=work_dir, file_name='analysis_parameters.json')
-        with open(file_path.path, 'w')  as outfile:
-            _dict = {k: v for k, v in query_dict.items() if v is not None}
-            my_json_str = json.dumps(_dict, indent=4, sort_keys=True)
-            outfile.write(u'%s' % my_json_str) # why is this formatting necessary?
+    def dump_analysis_parameters(self, work_dir, query_dict):
+        file_path = FilePath(file_dir=work_dir, file_name='analysis_parameters.json')
+        if not os.path.exists(file_path.path):
+            with open(file_path.path, 'w')  as outfile:
+                _dict = {k: v for k, v in query_dict.items() if v is not None}
+                my_json_str = json.dumps(_dict, indent=4, sort_keys=True)
+                outfile.write(u'%s' % my_json_str)
```

Comment:
no but really, why do we need here `u'%s'`?

Reply:
I did some test and if it is not formatted, then it has some issues during the extraction, but I didn't dig any further